test/rmui_bundle_suggestors/dart_script_adder_test.dart

```diff
@@ -0,0 +1,307 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:over_react_codemod/src/rmui_bundle_suggestors/constants.dart';
+import 'package:over_react_codemod/src/rmui_bundle_suggestors/dart_script_adder.dart';
+import 'package:test/test.dart';
+
+import '../util.dart';
+
+main() {
+  group('DartScriptAdder', () {
+    final testSuggestor = getSuggestorTester(DartScriptAdder(rmuiBundleScript));
+
+    test('empty file', () async {
+      await testSuggestor(expectedPatchCount: 0, input: '');
+    });
+
+    test('no matches', () async {
+      await testSuggestor(
+        expectedPatchCount: 0,
+        input: '''
+          final script = '<script src="/packages/react/react.js"></script>';
```

Comment:
`react.js` is left out of the regex in the ticket description of [CPLAT-15670](https://jira.atl.workiva.net/browse/CPLAT-15670), just want to verify that is intentional?

Reply:
The intent with that was to hopefully find the last one, but it was written a little hastily.

To include React, you can have either:
```html
<!-- 1. Dev react -->
<script src="/packages/react/react.js"></script>
<script src="/packages/react/react_dom.js"></script>

<!-- 2. Dev react with testing stuff (almost always used for tests) -->
<script src="/packages/react/react_with_addons.js"></script>
<script src="/packages/react/react_dom.js"></script>

<!-- 3. Prod react -->
<script src="/packages/react/react_prod.js"></script>
<script src="/packages/react/react_dom_prod.js"></script>

<!-- 4. Prod react, but one file -->
<script src="/packages/react/react_with_react_dom_prod.js"></script>
```

So I think as we're handling "2" correctly for this codemod, we should be good

Reply:
Oh okay that makes sense! Thank you for that explanation!

Reply:
Np! For the HTML file tests though, would you mind adding some test cases for those other cases?

Reply:
Added tests in 3241b8e (I ended up updating the regex to include `react.js` since the codemod already has logic to only add RMUI after the last js file)